### PR TITLE
Spiritball Update

### PR DESCRIPTION
**Pull request by MrKeiKun**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Follow-up to #3030. Merged both `clif_soulball` and `clif_spiritball` into one.

**Issues addressed:** 

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/map.h [hunk 1]
```diff
@@ -439,6 +439,16 @@ enum elements {
...
 };
 
+/**
+ * Types of Ball Types
+ * Used by clif_spiritball [KeiKun]
+ */
+enum spirit_ball_types {
+	BALL_TYPE_NONE = 0,
+	BALL_TYPE_SPIRIT,
+	BALL_TYPE_SOUL
```
Comment (4144) [inline src/map/map.h:449]: you sure it using names like this?

**Review by 4144 — CHANGES_REQUESTED** [review]

File: src/map/map.h (hunk 1 above)
Comment (MrKeiKun) [reply src/map/map.h:449]: I just used those to avoid using `int` as type. is there an official names for it?

Comment (4144) [reply src/map/map.h:449]: i cant find real enum, but is this can support ball type bigger than 2? is client draw it in some way?

**Comment by MrKeiKun** [comment]
will squash later after another HPM Hook Update~

**Review by 4144 — APPROVED** [review]

File: src/map/atcommand.c [hunk 2]
```diff
@@ -3501,7 +3501,7 @@ ACMD(spiritball)
 	if( sd->spiritball > 0 )
 		pc->delspiritball(sd, sd->spiritball, 1);
 	sd->spiritball = number;
-	clif->spiritball(&sd->bl, BALL_TYPE_SPIRIT, AREA);
+	clif->spiritball(&sd->bl, 1, AREA);
```
Comment (MishimaHaruna) [inline src/map/atcommand.c:3504]: I'm not really a fan of magic numbers, can you revert this to use an enum? It makes the code a lot easier to read

**Review by MishimaHaruna — CHANGES_REQUESTED** [review]

File: src/map/atcommand.c (hunk 2 above)
Comment (4144) [reply src/map/atcommand.c:3504]: this parameter set how many spirits should be spawned.
i don't think make any sense to use enum here.

Comment (MishimaHaruna) [reply src/map/atcommand.c:3504]: But then why does it say this

```
 * Ball Type:
 *      0 = None
 *      1 = Spirit
 *      2 = Soul
```

Comment (4144) [reply src/map/atcommand.c:3504]: ah here it not assigned directly to ``num`` then yes can be used enum

File: src/map/clif.c [hunk 3]
```diff
@@ -1593,9 +1593,9 @@ static bool clif_spawn(struct block_list *bl)
 			struct map_session_data *sd = BL_UCAST(BL_PC, bl);
 			int i;
 			if (sd->spiritball > 0)
-				clif->spiritball(&sd->bl);
+				clif->spiritball(&sd->bl, 1, AREA);
 			if (sd->soulball > 0)
-				clif->soulball(sd, NULL, AREA);
+				clif->spiritball(NULL, 2, AREA);
```
Comment (Lemongrass3110) [inline src/map/clif.c:1598]: You lost the &sd->bl here

**Pull request merged**